src/Stack/Dot.hs

```diff
@@ -70,16 +80,37 @@ dot :: (HasEnvConfig env
     => DotOpts
     -> m ()
 dot dotOpts = do
-    localNames <- liftM Map.keysSet getLocalPackageViews
-    resultGraph <- createDependencyGraph dotOpts
-    let pkgsToPrune = if dotIncludeBase dotOpts
-                         then dotPrune dotOpts
-                         else Set.insert "base" (dotPrune dotOpts)
-        prunedGraph = pruneGraph localNames pkgsToPrune resultGraph
-    printGraph dotOpts localNames prunedGraph
+  (localNames, prunedGraph) <- createPrunedDependencyGraph dotOpts
+  printGraph dotOpts localNames prunedGraph
+
+-- | Information about a package, when available.
+type Payload = (Maybe Version, Maybe License)
```

Comment:
No reason in particular. I’ll change it to a record type as that makes the code below more readable.